File: src/terceraParteEjerciciosTraduccion/ejercicio1/ejercicio1.java
package terceraParteEjerciciosTraduccion.ejercicio1;

import java.util.ArrayList;

//1. Sistema de Gestión de Biblioteca
// Crea un sistema de gestión de una biblioteca. Define clases para Libro, Usuario, y
// Biblioteca. Implementa métodos para agregar libros, registrar usuarios y prestar
// libros a los usuarios. Asegúrate de que un libro no se pueda prestar si ya está
// prestado.
import java.util.Scanner;

class Libro {
    String titulo;
    boolean prestado = false;

    public Libro(String titulo) {
        this.titulo = titulo;
    }

    public void prestar() {
        if (!prestado) {
            prestado = true;
            System.out.println("Libro prestado exitosamente.");
        } else {
            System.out.println("Este libro ya está prestado.");
        }
    }
}

class Usuario {
    String nombre;

    public Usuario(String nombre) {
        this.nombre = nombre;
    }
}

class Biblioteca {
    ArrayList<Libro> libros = new ArrayList<>();
    ArrayList<Usuario> usuarios = new ArrayList<>();

    public void agregarLibro(String titulo) {
        libros.add(new Libro(titulo));
        System.out.println("Libro agregado: " + titulo);
    }

    public void registrarUsuario(String nombre) {
        usuarios.add(new Usuario(nombre));
        System.out.println("Usuario registrado: " + nombre);
    }

    public void prestarLibro(String titulo) {
        for (Libro libro : libros) {
            if (libro.titulo.equalsIgnoreCase(titulo)) {
                libro.prestar();
                return;
            }
        }
        System.out.println("El libro no se encontró.");
    }
}

class Main {
    public static void main(String[] args) {
        Biblioteca2 biblioteca = new Biblioteca2();
        Scanner scanner = new Scanner(System.in);
        int opcion;

        do {
            System.out.println("\n1. Agregar libro");
            System.out.println("2. Registrar usuario");
            System.out.println("3. Prestar libro");
            System.out.println("4. Salir");
            System.out.print("Elige una opción: ");
            opcion = scanner.nextInt();
            scanner.nextLine();

            switch (opcion) {
                case 1:
                    System.out.print("Título del libro: ");
                    String titulo = scanner.nextLine();
                    biblioteca.agregarLibro(titulo);
                    break;
                case 2:
                    System.out.print("Nombre del usuario: ");
                    String nombre = scanner.nextLine();
                    biblioteca.registrarUsuario(nombre);
                    break;
                case 3:
                    System.out.print("Título del libro a prestar: ");
                    titulo = scanner.nextLine();
                    biblioteca.prestarLibro(titulo);
                    break;
                case 4:
                    System.out.println("Saliendo...");
                    break;
                default:
                    System.out.println("Opción no válida.");
            }
        } while (opcion != 4);

        scanner.close();
    }
}

File: src/terceraParteEjerciciosTraduccion/ejercicio1/ejercicio1.kt
package terceraParteEjerciciosTraduccion.ejercicio1

import kotlin.system.exitProcess

class Libro2(val titulo: String, var prestado: Boolean = false) {
    fun prestar() {
        if (!prestado) {
            prestado = true
            println("Libro prestado exitosamente.")
        } else {
            println("Este libro ya está prestado.")
        }
    }
}

class Usuario2(val nombre: String)

class Biblioteca2 {
    private val libros = mutableListOf<Libro2>()
    private val usuarios = mutableListOf<Usuario2>()

    fun agregarLibro(titulo: String) {
        libros.add(Libro2(titulo))
        println("Libro agregado: $titulo")
    }

    fun registrarUsuario(nombre: String) {
        usuarios.add(Usuario2(nombre))
        println("Usuario registrado: $nombre")
    }

    fun prestarLibro(titulo: String) {
        val libro = libros.find { it.titulo.equals(titulo, ignoreCase = true) }
        libro?.prestar() ?: println("El libro no se encontró.")
    }
}

fun main() {
    val biblioteca = Biblioteca2()
    while (true) {
        println("\n1. Agregar libro")
        println("2. Registrar usuario")
        println("3. Prestar libro")
        println("4. Salir")
        print("Elige una opción: ")
        val opcion = readlnOrNull()?.toIntOrNull() ?: 0

        when (opcion) {
            1 -> {
                print("Título del libro: ")
                val titulo = readlnOrNull() ?: ""
                biblioteca.agregarLibro(titulo)
            }
            2 -> {
                print("Nombre del usuario: ")
                val nombre = readlnOrNull() ?: ""
                biblioteca.registrarUsuario(nombre)
            }
            3 -> {
                print("Título del libro a prestar: ")
                val titulo = readlnOrNull() ?: ""
                biblioteca.prestarLibro(titulo)
            }
            4 -> {
                println("Saliendo...")
                exitProcess(0)
            }
            else -> println("Opción no válida.")
        }
    }
}
